components/calendar/src/types.rs

```diff
@@ -19,10 +19,111 @@ use zerovec::ule::AsULE;
+    /// Singleton Dangi era
+    pub const DANGI: Era = Era(tinystr!(16, "dangi"));
+    /// Incarnation (Ethiopian)
+    pub const INCARNATION: Era = Era(tinystr!(16, "incar"));
+    /// Pre-Incarnation (Ethiopian)
+    pub const PRE_INCARNATION: Era = Era(tinystr!(16, "pre-incar"));
+    /// Anno Mundi (Ethiopian)
+    pub const MUNDI: Era = Era(tinystr!(16, "mundi"));
+    /// Anno Mundi (Hebrew)
+    pub const AM: Era = Era(tinystr!(16, "am"));
+    /// Before Common Era (Julian, Gregorian)
+    pub const BCE: Era = Era(tinystr!(16, "bce"));
+    /// Common Era (Julian, Gregorian)
+    pub const CE: Era = Era(tinystr!(16, "ce"));
+    /// Anno Hegirae (Hijri, Persian)
+    pub const AH: Era = Era(tinystr!(16, "ah"));
+    /// Saka (Indian)
+    pub const SAKA: Era = Era(tinystr!(16, "saka"));
+    /// ISO-8601 default era
+    pub const DEFAULT: Era = Era(tinystr!(16, "default"));
+    /// Pre-ROC
+    pub const ROC_INVERSE: Era = Era(tinystr!(16, "roc-inverse"));
+    /// ROC
+    pub const ROC: Era = Era(tinystr!(16, "roc"));
+
+    /// Alias for [`Self::AD`]
+    pub const COPTIC: Era = Era(tinystr!(16, "coptic"));
+    /// Alias for [`Self::BD`]
+    pub const COPTIC_INVERSE: Era = Era(tinystr!(16, "coptic-inverse"));
+    /// Alias for [`Self::MUNDI`]
+    pub const ETHIOAA: Era = Era(tinystr!(16, "ethioaa"));
+    /// Alias for [`Self::INCARNATION`]
+    pub const ETHIOPIC: Era = Era(tinystr!(16, "ethiopic"));
+    /// Alias for [`Self::PRE_INCARNATION`]
+    pub const ETHIOPIC_INVERSE: Era = Era(tinystr!(16, "ethiopic-inverse"));
+    /// Alias for [`Self::AM`]
+    pub const HEBREW: Era = Era(tinystr!(16, "hebrew"));
+    /// Alias for [`Self::AH`]
+    pub const ISLAMIC_CIVIL: Era = Era(tinystr!(16, "islamic-civil"));
+    /// Alias for [`Self::AH`]
+    pub const ISLAMIC: Era = Era(tinystr!(16, "islamic"));
+    /// Alias for [`Self::AH`]
+    pub const ISLAMICC: Era = Era(tinystr!(16, "islamicc"));
+    /// Alias for [`Self::AH`]
+    pub const ISLAMIC_TBLA: Era = Era(tinystr!(16, "islamic-tbla"));
+    /// Alias for [`Self::AH`]
+    pub const ISLAMIC_UMALQURA: Era = Era(tinystr!(16, "islamic-umalqura"));
+    /// Alias for [`Self::SAKA`]
+    pub const INDIAN: Era = Era(tinystr!(16, "indian"));
+    /// Alias for [`Self::BCE`]
+    pub const JULIAN_INVERSE: Era = Era(tinystr!(16, "julian-inverse"));
+    /// Alias for [`Self::CE`]
+    pub const JULIAN: Era = Era(tinystr!(16, "julian"));
+    /// Alias for [`Self::AH`]
+    pub const PERSIAN: Era = Era(tinystr!(16, "persian"));
+
+    /// Heisei (Japanese)
+    pub const HEISEI: Era = Era(tinystr!(16, "heisei"));
+    /// Meiji (Japanese)
+    pub const MEIJI: Era = Era(tinystr!(16, "meiji"));
+    /// Reiwa (Japanese)
+    pub const REIWA: Era = Era(tinystr!(16, "reiwa"));
+    /// Showa (Japanese)
+    pub const SHOWA: Era = Era(tinystr!(16, "showa"));
+    /// Taisho (Japanese)
+    pub const TAISHO: Era = Era(tinystr!(16, "taisho"));
+
+    // output only?!
```

Comment:
These should be accepted on input too.

Reply:
I agree, hence the "?!", but I don't want to make changes in this PR.